zfit/minimizers/errors.py

```diff
@@ -0,0 +1,199 @@
+from ..param import set_values
+from ..util.container import convert_to_container
+
+
+class NewMinimum(Exception):
+    """Exception class for cases where a new minimum is found."""
+    pass
+
+
+def pll(minimizer, loss, params, values) -> float:
+    """Compute minimum profile likelihood for given parameters and values."""
+    params = convert_to_container(params)
+    values = convert_to_container(values)
+
+    verbosity = minimizer.verbosity
+    minimizer.verbosity = 0
+
+    with set_values(params, values):
+        for param in params:
+            param.floating = False
+        minimum = minimizer.minimize(loss=loss)
+
+    for param in params:
+        param.floating = True
+    minimizer.verbosity = verbosity
+
+    return minimum.fmin
+
+
+def set_params_to_result(params, result):
+    """Set parameters values to the values in the fitted value in the result."""
+    for param in params:
+        param.set_value(result.params[param]["value"])
+
+
+def get_crossing_value(result, params, direction, sigma, rootf, rtol):
+    """Find the crossing point between the profiled loss function, for given parameters, and the value of
+    `errordef` for a given direction (positive / negative).
+    `errordef` = 1 for a chisquare fit, = 0.5 for a likelihood fit.
+    """
+
+    all_params = list(result.params.keys())
+    loss = result.loss
+    errordef = loss.errordef
+    fmin = result.fmin
+    minimizer = result.minimizer.copy()
+    minimizer.tolerance = minimizer.tolerance * 0.5
+    rtol *= errordef
+
+    set_params_to_result(all_params, result)
+
+    covariance = result.covariance(as_dict=True)
+
+    step_size_dict = {ap: ap.step_size for ap in all_params}
+    for ap in all_params:
+        ap.step_size = covariance[(ap, ap)] ** 0.5
+
+    to_return = {}
+    for param in params:
+        param_error = result.hesse(params=param)[param]["error"]
+        param_value = result.params[param]["value"]
+        exp_root = param_value + sigma * direction * param_error  # expected root
+
+        for ap in all_params:
+            if ap == param:
+                continue
+
+            # shift parameters, other than param, using covariance matrix
+            ap_value = result.params[ap]["value"]
+            ap_error = covariance[(ap, ap)] ** 0.5
+            ap_value += sigma * direction * covariance[(param, ap)] / ap_error
+            ap.set_value(ap_value)
+
+        cache = {}
+        def shifted_pll(v):
+            """
+            Computes the pll, with the minimum substracted and shifted by minus the `errordef`, for a
+            given parameter.
+            `errordef` = 1 for a chisquare fit, = 0.5 for a likelihood fit.
+
+            The function raises a `NewMinimum` exception if the value of the shifted `pll` is less than
+            `- errordef`.
+            """
+            if v not in cache:
+                cache[v] = pll(minimizer, loss, param, v) - fmin - errordef
+                if cache[v] < -errordef:
```

Comment:
```suggestion
                if cache[v] < -errordef - minimizer.tolerance:
```
We don't want to have false positives, right? So a new minimum shouldn't be a fluctuation